Library/Homebrew/cmd/cleanup.rb

```diff
@@ -111,7 +111,11 @@ def cleanup_lockfiles
   end
 
   def rm_DS_Store
-    quiet_system "find", HOMEBREW_PREFIX.to_s, "-name", ".DS_Store", "-delete"
+    %w[Cellar Frameworks Library bin etc include lib opt sbin share var].each do |path|
+      path = HOMEBREW_PREFIX/path
+      next unless path.exist?
+      quiet_system "find", path.to_s, "-name", ".DS_Store", "-delete"
```

Comment:
> Why it will speed up? It's the same amount folders which will be scanned.

Yes, but it fires `find` only once. Each call to `quiet_system` does the following:

1. fork the current process
2. redirect `stdout` and `stderr` to `/dev/null`
3. `exec` the command
4. wait for the command to end

`find` is waaay faster than Ruby since it’s compiled from optimized C. I just tested the following on my machine:

```
# create 1000 directories with a lot of subdirectories
$ for i in {1..1000}; do d=dir$i/a/b/c/d/e/f/g/h/i/j/k/l/m/n/o/p/q/r; mkdir -p $d; touch $d/{x,y,z}; done

# search for all "x" files with `find`
$ time find dir{1..1000} -name x >/dev/null

real	0m0.272s
user	0m0.057s
sys	0m0.215s

# do the same in Ruby
$ brew irb
>>> b=Time.now;(1..1000).each { |i| quiet_system "find", "dir#{i}", "-name", "x" };Time.now-b
5.933559
```

That’s just a test, but it took 20+ times more time to do the loop in Ruby than letting `find` do it itself.

> This wouldn't work.

Right, sorry.